test/unit/host_test.rb

```diff
@@ -710,8 +710,6 @@ def teardown
       hostgroup = FactoryGirl.create(:hostgroup, :with_puppet_orchestration)
       h = FactoryGirl.create(:host, :managed, :with_environment, :hostgroup => hostgroup)
       Setting[:manage_puppetca] = true
-      assert h.puppet_proxy.present?
-      assert h.puppetca?
```

Comment:
This patch ought to improve the situation for Discovery in this regard, since inheriting these attributes is then the responsibility of the controller - you should be able to update it in the same way as core controllers.

If you have no concerns about this, I'll be merging it.